src/core/utils.js

```diff
@@ -494,6 +495,25 @@ export function flatten(collector, item) {
 
 // --- DOM HELPERS -------------------------------
 
+// Takes an array and returns a string that separates each of its items with the proper commas and
```

Comment:
I think there is some room for improvement in this comment.

Reply:
Agree, and this means our existing comment for `joinAnd` is also bad... 😱

Maybe some JSDoc will allow removing some prose.

Reply:
My suggestion:

```js
/**
 * Separates each item with proper commas and "and".
 * @param {string[]} array
 * @param {(str: string) => object} mapper
 */
```

Anything else looks redundant.

Reply:
LGTM.

Reply:
Suggested comment looks good.

Reply:
Shall I try using JSDoc

Reply:
Yup, https://github.com/w3c/respec/pull/2613#discussion_r356013231 uses JSDoc syntax. (Note the use of `/**` and `@param`.)